src/new-contract-interaction/agent.js

```diff
@@ -0,0 +1,198 @@
+const {
+  Finding, FindingSeverity, FindingType, getEthersProvider,
+} = require('forta-agent');
+
+// helper function to create alert for contract interaction
+function createContractInteractionAlert(
+  contractName,
+  contractAddress,
+  interactionAddress,
+  findingType,
+  findingSeverity,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation,
+) {
+  const finding = {
+    name: `${protocolName} New Contract Interaction`,
+    description: `The ${contractName} contract interacted with a new contract ${interactionAddress}`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-NEW-CONTRACT-INTERACTION`,
+    type: FindingType[findingType],
+    severity: FindingSeverity[findingSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      interactionAddress,
+    },
+  };
+
+  return Finding.fromObject(finding);
+}
+
+// helper function to create alert for EOA interaction
+function createEOAInteractionAlert(
+  contractName,
+  contractAddress,
+  interactionAddress,
+  transactionCount,
+  findingType,
+  findingSeverity,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation,
+) {
+  const finding = {
+    name: `${protocolName} New EOA Interaction`,
+    description: `The ${contractName} contract interacted with a new EOA ${interactionAddress}`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-NEW-EOA-INTERACTION`,
+    type: FindingType[findingType],
+    severity: FindingSeverity[findingSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      interactionAddress,
+      transactionCount,
+    },
+  };
+
+  return Finding.fromObject(finding);
+}
+
+const initialize = async (config) => {
+  let agentState = {};
+  agentState.provider = getEthersProvider();
+
+  agentState.contractInfo = config.contracts;
+  agentState.developerAbbreviation = config.developerAbbreviation;
+  agentState.protocolName = config.protocolName;
+  agentState.protocolAbbreviation = config.protocolAbbreviation;
+
+  agentState.contracts = Object.entries(agentState.contractInfo).map(([name, entry]) => {
+    const {
+      thresholdBlockCount,
+      thresholdTransactionCount,
+      address,
+      filteredAddresses,
+      findingType,
+      findingSeverity,
+    } = entry.newContractEOA;
+
+    const contract = {
+      name,
+      address,
+      filteredAddresses,
+      thresholdBlockCount,
+      thresholdTransactionCount,
+      findingType,
+      findingSeverity,
+    };
+
+    return contract;
```

Comment:
Code cleanup from original implementation.
```suggestion
    return {
      name,
      ...entry.newContractEOA,
    };
```